docs/openapi/howto/binary-payload.md

```diff
@@ -0,0 +1,228 @@
+# How to represent binary/file payload in openapi
+
+This document will explain how you can represent various binary payload in swagger/openapi.
+
+## Basics
+
+There is 2 part to describe a binary/file payload:
+
+- Describe the `content-type`, this could be anything, for example
+
+  - `application/octet-stream`: For a raw binary file.
+  - `image/png`: To describe returning a png file.
+  - `application/json`: To describe returning some json file.(This could be used to describe a json file where the server/sdk has no control over the content)
+
+- Describe the schema of the payload. This is the important part to mark the payload as a binary/file. It is a different process for Swagger 2 and OpenAPI 3
+  - OpenAPI3: Use schema `{"type": "string", "format": "binary"}`
+  - Swagger: This one is a little more complex, for responses: `{"type": "file"}` and for requests `{"type": "string", "format": "binary"}` (This is a autorest specific feature. Read [below](#swagger-2-1) for more details.)
+
+[See more information on Swagger docs](https://swagger.io/docs/specification/describing-responses/#response-that-returns-a-file)
+
+### Response
+
+The overall response should look like this(Replace `{{INSERT_CONTENT_TYPE}}` with desired content type):
+
+#### OpenAPI 3
+
+```json
+{
+  "paths": {
+    "/return-file": {
+      "get": {
+        "responses": {
+          "200": {
+            "description": "Returns a binary file",
+            "content": {
+              "{{INSERT_CONTENT_TYPE}}": {
+                "schema": {
+                  "type": "string",
+                  "format": "binary"
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+#### Swagger 2
+
+```json
+{
+  "paths": {
+    "/return-file": {
+      "get": {
+        "produces": ["{{INSERT_CONTENT_TYPE}}"],
+        "responses": {
+          "200": {
+            "description": "Returns a binary file",
+            "schema": {
+              "type": "file"
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+### Request body
```

Comment:
nit: I think we should go request body then response body

Reply:
Yeah I agree it makes more sense, just did that because the request body was a little more special for swagger